app/workers/postcodes_collection_worker.rb

```diff
@@ -0,0 +1,13 @@
+class PostcodesCollectionWorker < ApplicationWorker
+  sidekiq_options queue: :collect_postcodes
+
+  def perform
+    run_with_advisory_lock(Postcode, postcode) do
+      postcodes = Postcode.select(:postcode)
+
+      postcodes.each do |postcode|
```

Comment:
When all the postcodes have been updated once, that's the end of this worker's work. As far as I can see, we don't yet have a mechanism for forcing another refresh. I think we'll need to wrap the contents of our `perform` method with a `while true`, i.e. when the `postcodes.each` iteration comes to an end, we'll want to run it all over again.

 We'll want a test for that too. Therefore probably worth doing in a second commit.